runtime/src/main/java/org/corfudb/runtime/view/TableRegistry.java

```diff
@@ -0,0 +1,170 @@
+        this.registryTable = this.runtime.getObjectsView().build()
+                .setTypeToken(new TypeToken<CorfuTable<TableName, CorfuRecord<TableDescriptors, Message>>>() {
+                })
+                .setStreamName(getFullyQualifiedTableName(CORFU_SYSTEM_NAMESPACE, REGISTRY_TABLE_NAME))
+                .setSerializer(this.protobufSerializer)
+                .open();
+
+        // Register the table schemas to schema table.
+        addTypeToClassMap(TableName.getDefaultInstance());
+        addTypeToClassMap(TableDescriptors.getDefaultInstance());
+
+        registerTable(CORFU_SYSTEM_NAMESPACE,
+                REGISTRY_TABLE_NAME,
+                TableName.getDescriptor().toProto(),
+                TableDescriptors.getDescriptor().toProto());
+    }
+
+    private void registerTable(String namespace,
+                               String tableName,
+                               DescriptorProto keyDescriptor,
+                               DescriptorProto valueDescriptor) {
+
+        TableName tableNameKey = TableName.newBuilder()
+                .setNamespace(namespace)
+                .setTableName(tableName)
+                .build();
+        TableDescriptors tableDescriptors = TableDescriptors.newBuilder()
+                .setKeyDescriptor(keyDescriptor)
+                .setValueDescriptor(valueDescriptor)
+                .build();
+
+        try {
+            this.runtime.getObjectsView().TXBuild().type(TransactionType.OPTIMISTIC).build().begin();
+            this.registryTable.putIfAbsent(tableNameKey,
+                    new CorfuRecord<>(tableDescriptors, null));
+        } finally {
+            this.runtime.getObjectsView().TXEnd();
+        }
+    }
+
+    private String getTypeUrl(Descriptor descriptor) {
+        return "type.googleapis.com/" + descriptor.getFullName();
+    }
+
+    private String getFullyQualifiedTableName(String namespace, String tableName) {
+        return namespace + "$" + tableName;
+    }
+
+    private <T extends Message> void addTypeToClassMap(T msg) {
+        String typeUrl = getTypeUrl(msg.getDescriptorForType());
+        // Register the schemas to schema table.
+        if (!classMap.containsKey(typeUrl)) {
+            classMap.put(typeUrl, msg.getClass());
+        }
+    }
+
+    public <K extends Message, V extends Message, M extends Message>
+    Table<K, V, M> openTable(@Nonnull final String namespace,
+                             @Nonnull final String tableName,
+                             @Nonnull final Class<K> kClass,
+                             @Nonnull final Class<V> vClass,
+                             @Nullable final Class<M> mClass,
+                             @Nonnull final TableOptions tableOptions)
+            throws NoSuchMethodException, InvocationTargetException, IllegalAccessException {
+
+        // Register the schemas to schema table.
+        K defaultKeyMessage = (K) kClass.getMethod("getDefaultInstance").invoke(null);
```

Comment:
why just not to pass a default instance?

Reply:
Just want to reduce the footprint of the create. We do need the actual type for the table, now we will need to ask the default instance of it as well. It makes the caller a lot more verbose.